src/fswAlgorithms/orbitControl/lambertSolver/lambertSolver.cpp

```diff
@@ -122,6 +122,51 @@ void LambertSolver::problemGeometry()
     this->O_frame2 = {i_r2, i_t2, i_h};
 }
 
+/*! This method finds the free variable x that satisfies the requested time of flight TOF.
+    @return void
+*/
+void LambertSolver::findx()
+{
+    // get initial guess for iteration variable x
+    std::vector<double> x0 = getInitialGuess(this->lambda, this->TOF);
+    if (this->M > 0 && !this->multiRevSolution){
+        // if getInitialGuess function determines that TOF is too short for multi-revolution solution to exist, return no solution
+        return;
+    }
+
+    // find x that satisfies time-of-flight constraint using numerical root finder
+    if (this->solverName == "Gooding"){
```

Comment:
:small_red_triangle_down: Same as before. Let's split these out to a gooding method and an izzo method. This now allows us to localize the logic for switching between methods to one place in the module. Makes it easier to follow, maintain, and test.

Reply:
Done.